Plot Regrets [Resolves #11]

thcrock opened the pull request:
- Create SelectionRulePlotter that creates a cumulative distribution plot of selection rules and their regrets
- Implement selection rule grid to efficiently create many bound selection rules

shaycrk commented on audition/regrets.py:
```diff
@@ -55,13 +57,132 @@ def regrets_for_rule(
+            title=plt_title,
+            x_label='decrease in {} next time from best model'.format(regret_metric),
+            y_label='fraction of models',
+            x_ticks=numpy.arange(0, 1.1, 0.1)
```
These ticks assume a 0-1 range, right? Also, the x-label of `decrease...` should probably be something like `distance from best`

thcrock replied:
Thanks, I just uploaded a change to use the same ticks that we already computed (as well as changing the copy)

shaycrk commented:
Running through the code and poking at this in a notebook, looks pretty good to me.

shaycrk commented:
Just getting around to testing this as well and noticed two things here:
* We should use the "distance from best" rather than "decrease in" language on the axis label for the plots looking at the model groups as well as the regret plots.
* Just using the same ticks as we're using to calculate the points for the plot is creating a tick density that's too high to be legible (see image below)

![image](https://user-images.githubusercontent.com/960183/29625690-0c736ac6-87f2-11e7-8272-417c73ea6151.png)

thcrock committed:
Plot Regrets [Resolves #11]

- Create SelectionRulePlotter that creates a cumulative distribution plot of selection rules and their regrets
- Implement selection rule grid to efficiently create many bound selection rules
